discovery-provider/src/tasks/entity_manager/entities/social_features.py

```diff
@@ -296,7 +296,7 @@ def validate_duplicate_social_feature(
     # Cannot duplicate a social feature
     key = get_record_key(params.user_id, params.entity_type, params.entity_id)
 
-    existing_record = params.existing_records.get(record_type, {}).get(key)
+    existing_record = cast(dict, params.existing_records).get(record_type, {}).get(key)
```

Comment:
huh. why does this need a cast I wonder? It's already a typeddict

Reply:
The types in this file are all a lie. It has to do with the Enum vs. String situation. I kind want to leave this to a future endeavor, but for now, I've added a Literal['Playlist', 'Track', ....] to fix the need for a cast.

There are some oopsie files around, e.g.
https://github.com/AudiusProject/audius-protocol/blob/65b609994359f0db816554136bc971a3605d9ee8/discovery-provider/src/tasks/entity_manager/utils.py#L195